### DISCUSSION TITLE
Confusing options for DFT-D4 in VASP

### DISCUSSION THREAD

**Discussion by LePingKYXK** [dftd4/dftd4#147] [Q&A]
Hi Sebastian,

I have compiled DFT-D4 with meson and ninja, and then I would like to link DFT-D4 to VASP 6.3.0. However, I found that the option provided by [VASP wiki](https://www.vasp.at/wiki/index.php/Makefile.include#DFTD4_.28optional.29) is 

```bash
CPP_OPTIONS += -DDFTD4
DFTD4_ROOT  ?= /path/to/your/dft4/installation
LLIBS       += -L$(DFTD4_ROOT)/build -ldftd4
INCS        += -I$(DFTD4_ROOT)/libdftd4.a.p
```

which is a little bit different from 
```bash
CPP_OPTIONS += -DDFTD4
LLIBS       += $(shell pkg-config --libs dftd4)
INCS        += $(shell pkg-config --cflags dftd4)
```

Do I need to specify the DFTD4_ROOT in the **makefile.include** when compiling VASP 6.3.0?

Best regards,
Huan

**Comment by awvwgk** [comment]
The options in the Vasp wiki assume you will compile against the build directory, which we discourage because the internal structure of this directory is not really stable. The `DFTD4_ROOT` is mainly for setting the prefix in the manual example.

However this is not required when getting the information from pkg-config, which is the preferred way to build with support for DFT-D4.

**Reply by LePingKYXK** [reply]
Hi Sebastian,

Thank you very much for your quick response. I will follow your method by using pkg-config.

**Reply by awvwgk** [reply]
Is it possible to update the build instructions in the Vasp wiki? I would much appreciate if they match our recommended approach.

**Reply by LePingKYXK** [reply]
I think it is possible to update the compiling instruction by posting this issue to the [VASP Forums](https://www.vasp.at/forum/viewforum.php?f=2). However, I could not find the way to register... 
I have sent a message asking for registration to the administrator via `contact us`, and waiting for the response.
